routing/dht/handlers.go

```diff
@@ -226,5 +226,5 @@ func (dht *IpfsDHT) handleAddProvider(ctx context.Context, p peer.ID, pmes *pb.M
 		dht.providers.AddProvider(ctx, key, p)
 	}
 
-	return pmes, nil // send back same msg as confirmation.
```

Comment:
This change makes it more closely follow kademlia, and makes more sense with us wanting to change over to a UDP protocol for the DHT (not saying thats happening soon, just those concepts).